src/v/kafka/server/server.cc

```diff
@@ -874,6 +874,19 @@ add_offsets_to_txn_handler::handle(request_context ctx, ss::smp_service_group) {
             add_offsets_to_txn_response response;
             response.data.error_code = error_code::policy_violation;
             return ctx.respond(response);
+        } else if (!ctx.authorized(
```

Comment:
So we won't return error if you are either one of authorized to read or write, but are there any checks further in the the request handler to ensure that if you are doing an action of read you didn't pass through this area when you were authorized to write?

Reply:
and viceversa

Reply:
These are cribbed directly (though by hand, so not infallibly!) from mainline kafka. In plain english, to add offsets to the transaction, you need WRITE permissions for the transactional ID **and** READ permissions for the group. 

To be perfectly honest I'm not 100% on the semantics here, but, to my understanding, the request is ill-formed if those conditions aren't met.

Reply:
This seems consistent with Apache Kafka code.

> but are there any checks further in the the request handler to ensure that if you are doing an action of read you didn't pass through this area when you were authorized to write?

I'm not sure I followed this.